### PR TITLE
Change ip_neighbor module replaced state design

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
This is to re-design ip_neighbor module "replaced" state according to newly specified "replaced" definition.

GitHub Issues
N/A

ISSUE TYPE
- Feature Pull Request
- 
COMPONENT NAME
sonic_ip_neighbor

ADDITIONAL INFORMATION
- regression result: 
[regression-2023-05-01-10-17-42.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11366895/regression-2023-05-01-10-17-42.html.pdf)
- regression log: 
[regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11366899/regression_test.log)

 Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

How Has This Been Tested?
Tested with real SONIC systems.

### CODE REVIEW

**Review by stalabi1 — APPROVED** [review]

File: plugins/module_utils/network/sonic/config/ip_neighbor/ip_neighbor.py
Comment (kerry-meyer) [inline plugins/module_utils/network/sonic/config/ip_neighbor/ip_neighbor.py:1]: These changes look good, but please also add the following change:

With these changes, the "_state_overridden" and "state_replaced" functions no longer use the input "diff" parameter. Please delete this input from both of those functions.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]

File: plugins/module_utils/network/sonic/config/ip_neighbor/ip_neighbor.py
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/ip_neighbor/ip_neighbor.py:1]: Changed. Thanks!

Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/ip_neighbor/ip_neighbor.py:1]: The updates look fine.

Thanks for making these additional changes.

I will "Resolve" this issue and Approve/merge this PR.

**Review by kerry-meyer — APPROVED** [review]
Thanks for making the additional changes.

Approved.

**Pull request merged**